### PR TITLE
Add a helper to enable use of compiled LSRK45

### PR DESCRIPTION
**Pull request by inducer**
This seems to work, and it seems to help a fair bit for the wave operator. It's at 0.008s on my laptop. (Earlier measurements were 0.06s, but I forgot what those were.)

Draft because:
- [x] Needs https://github.com/inducer/pytato/pull/308
- [x] ~~Find a way to get rid of the extra `_cls_has_array_context_attr=True`.~~ Can't, currently. See discussion in https://github.com/inducer/arraycontext/pull/155.
- [x] Wait for https://github.com/inducer/arraycontext/pull/155 to make this easier to use.
- [x] Use this in a convergence test.

cc @MTCam

### CODE REVIEW

**Comment by inducer** [comment]
This currently rolls over when trying to use the fusion array contractor with
```
NotImplementedError: Cannot fit loop nest 'frozenset()' into known set of loop-nest patterns.
```
cc @kaushikcfd

**Comment by kaushikcfd** [comment]
A temporary workaround for that would be:
```patch
diff --git a/meshmode/array_context.py b/meshmode/array_context.py
index e155671..e9fa6e0 100644
--- a/meshmode/array_context.py
+++ b/meshmode/array_context.py
@@ -1272,6 +1272,19 @@ class FusionContractorArrayContext(
 
         # }}}
 
+        # {{{ un-materialize scalars
+
+        def unmaterialize_scalars(expr):
+            if isinstance(expr, pt.Array) and expr.shape == ():
+                return expr.without_tags(pt.tags.ImplStored(),
+                                         verify_existence=False)
+            else:
+                return expr
+
+        dag = pt.transform.map_and_copy(dag, unmaterialize_scalars)
+
+        # }}}
+
         # {{{ infer axis types
 
         from meshmode.pytato_utils import (unify_discretization_entity_tags,
```

This blatant un-materialization isn't great. But I can't think of a general way to handle this pattern.

**Comment by inducer** [comment]
```
residual = 0*y
```
Applied.

**Comment by inducer** [comment]
There's still a failure here when running distributed:
```
pytools.tag.NonUniqueTagError: Multiple tags are direct subclasses of the following UniqueTag(s): PartIDTag
```
I attribute that to https://github.com/inducer/pytato/issues/307.

File: examples/wave/wave-op-mpi.py [hunk 1]
```diff
@@ -57,7 +57,9 @@
 
 # {{{ wave equation bits
 
-@with_container_arithmetic(bcast_obj_array=True, rel_comparison=True)
+@with_container_arithmetic(bcast_obj_array=True, rel_comparison=True,
+        # FIXME: I'll put together an actx patch to make this unnecessary
+        _cls_has_array_context_attr=True)
```
Comment (inducer) [inline examples/wave/wave-op-mpi.py:62]: We can delete this once https://github.com/inducer/arraycontext/pull/155 is in.

Comment (inducer) [reply examples/wave/wave-op-mpi.py]: Alas, it is not to be. See the discussion in https://github.com/inducer/arraycontext/pull/155.

**Comment by inducer** [comment]
The latest timings here: Without this, 0.0164s per step, with this change 0.0078s per step. There is another usability issue that I'd like to get to before merging this (can't currently pass numpy array scalars to compiled functions). Beyond that, there are quite a few more low-hanging fruit.

**Comment by inducer** [comment]
> There is another usability issue that I'd like to get to before merging this (can't currently pass numpy array scalars to compiled functions).

I've decided that the run time cost of supporting this is too large. We'll have to burden the user with getting rid of array scalars beforehand. (In particular, `numpy.ndarray` satisfies `is_array_container_type`, so "detect array scalars" would be living in a very hot code path.)

**Pull request merged**